autogluon/utils/tabular/ml/models/lr/lr_model.py

```diff
@@ -0,0 +1,206 @@
+        logger.log(15, " - input shape %s" % str(X.shape))
+        if is_train:
+            transformer_list = []
+
+            if len(feature_types['language']) > 0:
+                pipeline = Pipeline(steps=[
+                    ("preparator", NlpDataPreprocessor(nlp_cols=feature_types['language'])),
+                    ("vectorizer",
+                     TfidfVectorizer(ngram_range=self.params['proc.ngram_range'], sublinear_tf=True, max_features=vect_max_features, tokenizer=self.tokenize))
+                ])
+                transformer_list.append(('vect', pipeline))
+
+            if len(feature_types['onehot']) > 0:
+                pipeline = Pipeline(steps=[
+                    ('generator', OheFeaturesGenerator(cats_cols=feature_types['onehot'])),
+                ])
+                transformer_list.append(('cats', pipeline))
+
+            if len(feature_types['continuous']) > 0:
+                pipeline = Pipeline(steps=[
+                    ('generator', NumericDataPreprocessor(cont_cols=feature_types['continuous'])),
+                    ('imputer', SimpleImputer(strategy=self.params['proc.impute_strategy'])),
+                    ('scaler', StandardScaler())
+                ])
+                transformer_list.append(('cont', pipeline))
+
+            if len(feature_types['skewed']) > 0:
+                pipeline = Pipeline(steps=[
+                    ('generator', NumericDataPreprocessor(cont_cols=feature_types['skewed'])),
+                    ('imputer', SimpleImputer(strategy=self.params['proc.impute_strategy'])),
+                    ('quantile', QuantileTransformer(output_distribution='normal')),  # Or output_distribution = 'uniform'
+                ])
+                transformer_list.append(('skew', pipeline))
+
+            self.pipeline = FeatureUnion(transformer_list=transformer_list)
+            logger.log(15, " - fitting pre-processing pipeline")
+            self.pipeline.fit(X)
+
+        logger.log(15, " - transforming inputs using pre-processing pipeline")
+        X = self.pipeline.transform(X)
+        logger.log(15, " - output shape %s" % str(X.shape))
+
+        return X
+
+    def _set_default_params(self):
+        default_params = {
+            'model_type': 'LR',
+            'C': 1,
+            'vectorizer_dict_size': 75000,
+            'proc.ngram_range': (1, 5),
+            'proc.skew_threshold': 0.99,
+            'proc.impute_strategy': 'median',  # strategy argument of sklearn.SimpleImputer() used to impute missing numeric values
+        }
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_searchspace(self, problem_type):
+        spaces = {
+            # 'C': Real(lower=1e-4, upper=1e5, default=1),
+            # 'tokenizer': Categorical('split', 'sentencepiece')
+        }
+        return spaces
+
+    def fit(self, X_train, Y_train, X_test=None, Y_test=None, time_limit=None, **kwargs):
```

Comment:
Remove X_train, Y_train, and time_limit args, since they are never used.

Reply:
It could be possible to adaptively set `max_iter`:  https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#examples-using-sklearn-linear-model-logisticregression

to approximately respect `time_limit` based on sample-size, feature-dimensionality, and the `solver` used.

Reply:
Probably not necessary for first version but definitely something to keep in mind

Reply:
Added TODO for this version